game.py

```diff
@@ -12,15 +12,46 @@
     '-----------'
 ]
 
-
-def snowman(snowman_word):
+def print_snowman(wrong_guesses_count):
+    """Prints the snowman graphic based on wrong guesses."""
+    for i in range(SNOWMAN_MAX_WRONG_GUESSES - wrong_guesses_count, SNOWMAN_MAX_WRONG_GUESSES):
+        print(SNOWMAN_GRAPHIC[i])
+    
+def snowman(snowman_word=None):
     """Complete the snowman function
     replace "pass" below with your own code
     It should print 'Congratulations, you win!'
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    # Initialize the correct_letter_guess_statuses dictionary
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+
+    # Loop until the player wins or loses
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES:
+        print("\nWord: " + generate_word_progress_string(snowman_word, correct_letter_guess_statuses))
```

Comment:
This line breaks the PEP8 guidelines for line length (79 characters or less including indentation https://peps.python.org/pep-0008/#maximum-line-length). From the linked guide:

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses.

I'll be honest, this is a tougher line to break up nicely, but you may be able to get away with something like what I've added below:

```suggestion
        print("\nWord: " + 
             generate_word_progress_string(snowman_word, correct_letter_guess_statuses))
```
or 

```suggestion
        print("\nWord: " + generate_word_progress_string(snowman_word, 
             correct_letter_guess_statuses))
```